server/controllers/request.js

```diff
@@ -0,0 +1,75 @@
+require('dotenv').config();
+const Request = require('../models/Request');
+const asyncHandler = require('express-async-handler');
+
+// @route GET /request
+// @desc get requests related to a logged in user or sitter
+// @access Private
+exports.requestGet = asyncHandler(async (req, res) => {
+  const userId = req.user.id;
+
+  const getUser = await Request.find({userId: userId})
+  //
+  if(getUser != null){
+    res.send(200).json({ success:{ message:  "get the " + getUser }});
+  } else {
+    res.status(500).send("user didn't exist.");
+  }
+
+});
+
+
+
+
+// @route POST /request
+// @desc post requests with user id and sitter id
+// @access Private
+exports.requestPost = asyncHandler(async (req, res) => {
+  const { sitterId, start, end, offset } = req.body;
+
+  const userId = req.user.id;
+
+  const getUser = await Request.find({userId: userId});
+
+  if(getUser != null){
+    res.status(500).send("user is exist, can not be dupicate");
```

Comment:
Wrong status code. Error message is mispelled too. Not sure why you are preventing multiple requests?

Reply:
I want to make sure userId is unique, userId and sitterId are unique

Reply:
Users can create as many requests for a pet sitter as they need. What would be more ideal is ensuring the pet sitter id exists instead

Reply:
The route is protected - only authenticated users can access it. So we can be confident that the user does exist but not confident the pet sitter always exists.